thoth-tech/doubtfire-web
# Migrate/comments-modal component

Student ID: s222313935

# Description

Migrates the `comments-modal` component from CoffeeScript to TypeScript. This component is used to display images and PDFs from the task comments as a fullscreen modal popup, allowing users to view the content without needing to download or open it in another tab.

## Type of change


# How Has This Been Tested?
- Log in as `student_1` and select any unit and any task.
- Send an image and a PDF into the task comments either by manually attaching the files using the bottom left icon or dragging and dropping it over.
- Once uploaded, confirm the `comments-modal` is working correctly by clicking on the attachments in the comments and view images in fullscreen or PDFs embedded into the modal.

## Before (images):
![comments-modal-image-original](https://github.com/user-attachments/assets/c20fbf99-4637-40ba-96e6-519958f7fcbc)

## After (images):
![comments-modal-image-migrated](https://github.com/user-attachments/assets/48e7c756-38df-4c3e-a6a2-925347816788)

## Before (PDFs):
![comments-modal-pdf-original](https://github.com/user-attachments/assets/739a5768-c679-45a4-8a7a-8b5e284d70b5)

## After (PDFs):
![comments-modal-pdf-migrated](https://github.com/user-attachments/assets/85618591-6b7a-4dcd-a862-5b873fb00e33)

## Testing Checklist:


# Checklist:


Component review PR: https://github.com/thoth-tech/documentation/pull/519
